### PR TITLE
Use react-native-fast-image for remote images

### LINKED ISSUE #7633: Use react-native-fast-image for remote images
In stickers and in extensions we load remote images and react native image works not perfect, react-native-fast-image has much better performance and better to use it for remote images

https://github.com/DylanVann/react-native-fast-image

### PR DESCRIPTION
**Pull request by shamardy**
Fixes #7633

### CODE REVIEW

**Comment by flexsurfer** [comment]
hey @shamardy is it WIP?

**Comment by shamardy** [comment]
> hey @shamardy is it WIP?

No. Also I dont have Overall/Read permission in Jenkins to check the logs and errors.

**Comment by flexsurfer** [comment]
but i couldn't find any usage of this new component in the code, we should replace react native image on this new component

File: react-native/src/desktop/status_im/react_native/js_dependencies.cljs [hunk 1]
```diff
@@ -26,6 +26,7 @@
 (def status-keycard         (fn [] #js {:default #js {}}))
 (def dialogs                (fn [] #js {}))
 (def dismiss-keyboard       (fn [] #js {}))
+(def fast-image             (fn [] #js {}))
```
Comment (flexsurfer) [inline react-native/src/desktop/status_im/react_native/js_dependencies.cljs:30]: i'm not sure about desktop, @vkjr ?

File: src/status_im/extensions/capacities/components.cljs [hunk 2]
```diff
@@ -51,7 +52,7 @@
                                         (when style {:style style}))] children))
 
 (defn image [{:keys [source uri style]}]
-  [react/image (merge {:style (merge {:width 100 :height 100} style)} {:source (if source source {:uri uri})})])
+  [rn-dependencies/fast-image (merge {:style (merge {:width 100 :height 100} style)} {:source (if source source {:uri uri})})])
```
Comment (flexsurfer) [inline src/status_im/extensions/capacities/components.cljs]: we need to replace usage of class inside react/image itself, here https://github.com/status-im/status-react/blob/c2b55233225f6e4d59c9ca5174b5261cc1137fec/components/src/status_im/ui/components/react.cljs#L78, thanks

**Review by flexsurfer — CHANGES_REQUESTED** [review]
thanks for the contribution @shamardy left some comments

File: components/src/status_im/ui/components/react.cljs [hunk 3]
```diff
@@ -61,7 +61,7 @@
 
 (def text-class ((get-class "Text")))
 (def text-input-class (get-class "TextInput"))
-(def image-class (get-class "Image"))
+(def fast-image-class (get-class "fast-image"))
```
Comment (flexsurfer) [inline components/src/status_im/ui/components/react.cljs]: `get-class` retrieves class from react native package, but you need to get it from react-native-fast-image,  see how it works for image picker for example https://github.com/status-im/status-react/blob/f3aa376d167ed7d7dd5073b83f641cba7f861b03/components/src/status_im/ui/components/react.cljs#L202-L213

**Comment by flexsurfer** [comment]
hey @shamardy thanks for your effort, there is an error in ios build `[!] No podspec found for react-native-fast-imag in ../node_modules/react-native-fast-image`

File: react-native/src/desktop/status_im/react_native/js_dependencies.cljs [hunk 4]
```diff
@@ -26,6 +26,7 @@
 (def status-keycard         (fn [] #js {:default #js {}}))
 (def dialogs                (fn [] #js {}))
 (def dismiss-keyboard       (fn [] #js {}))
+(def fast-image             #js {:FastImage #js {}})
```
Comment (yenda) [inline react-native/src/desktop/status_im/react_native/js_dependencies.cljs]: Shouldn't desktop at least back up on standard image?

Comment (flexsurfer) [reply react-native/src/desktop/status_im/react_native/js_dependencies.cljs]: @vkjr ?

Comment (vkjr) [reply react-native/src/desktop/status_im/react_native/js_dependencies.cljs]: Yes, for desktop we should use default images until react-native-fast-image ported to desktop (if there is a need)

Comment (flexsurfer) [reply react-native/src/desktop/status_im/react_native/js_dependencies.cljs]: @shamardy could you take a look please, thanks

**Comment by flexsurfer** [comment]
hey @shamardy any updates?

**Comment by flexsurfer** [comment]
@shamardy could you rebase onto develop please, thanks

**Comment by oskarth** [comment]
Closing PR as no longer active. Please re-open if this is in error.

**Pull request closed without merging**